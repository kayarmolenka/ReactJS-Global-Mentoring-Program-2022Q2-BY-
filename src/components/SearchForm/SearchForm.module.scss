@import "src/style/variables";
@import "src/style/mixins";

.searchFormWrapper {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  h1 {
    @include font-family;
    font-weight: 300;
    font-size: 40px;
    line-height: 49px;
    letter-spacing: 1px;
    text-transform: uppercase;
    color: $white;
    padding-right: 981px;
  }

  .searchForm {
    flex-wrap: wrap;
    display: flex;

    .searchFormInput {
      @include font-family;
      background: rgba(50, 50, 50, 0.8);
      mix-blend-mode: normal;
      opacity: 0.7;
      border-radius: 4px;
      font-weight: 400;
      font-size: 20px;
      line-height: 24px;
      color: $white;
      padding: 0 800px 0 19px;
      height: 53px;
      min-width: 300px;
    }
    .searchFormBtn {
      @include font-family;
      color: $white;
      text-transform: uppercase;
      font-weight: 500;
      font-size: 20px;
      line-height: 24px;
      text-align: center;
      background: $pink;
      border-radius: 4px;
      padding: 18px 73px 15px 74px;
      cursor: pointer;
      margin-left: 14px;
      display: flex;
      justify-content: center;
      border: none;
    }
  }
}



