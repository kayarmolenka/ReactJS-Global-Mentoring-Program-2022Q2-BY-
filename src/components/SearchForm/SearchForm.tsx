import { ChangeEvent, useState } from 'react';
import { Button } from '../Button';
import { ButtonType } from '../../models';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faXmark } from '@fortawesome/free-solid-svg-icons';

import styles from './SearchForm.module.scss';

export const SearchForm = () => {
  const searchHandle = () => {
    console.log('searchHandle');
  };

  const [isVisibleCross, setIsVisibleCross] = useState(false);
  const [valueSearchForm, setValueSearchForm] = useState('');

  const handleValueSearchForm = (e: ChangeEvent<HTMLInputElement>) => {
    setValueSearchForm(e.target.value);
    if (e.target.value.length) {
      setIsVisibleCross(true);
    }
  };

  const removeSearchText = () => {
    setValueSearchForm('');
    setIsVisibleCross(false);
  };

  return (
    <div className={styles.searchFormWrapper}>
      <h1>Find your movie</h1>
      <div className={styles.searchForm}>
        <div>
          <input
            placeholder="What do you want to watch?"
            className={styles.searchFormInput}
            type="text"
            value={valueSearchForm}
            onChange={handleValueSearchForm}
          />
          {isVisibleCross && valueSearchForm.length && (
            <FontAwesomeIcon
              icon={faXmark}
              className={styles.searchFormIcon}
              onClick={removeSearchText}
            />
          )}
        </div>
        <Button
          text="Search"
          onClick={searchHandle}
          type={ButtonType.SUBMIT}
          className={styles.searchFormBtn}
        />
      </div>
    </div>
  );
};
