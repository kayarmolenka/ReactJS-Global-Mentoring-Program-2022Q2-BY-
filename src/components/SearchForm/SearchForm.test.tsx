import { render } from '@testing-library/react';
import { SearchForm } from './SearchForm';

describe('SearchForm', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<SearchForm />);

    expect(asFragment()).toMatchSnapshot();
  });
});
