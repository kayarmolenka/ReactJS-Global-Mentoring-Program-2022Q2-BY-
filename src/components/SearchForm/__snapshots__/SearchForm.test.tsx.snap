// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match snapshot 1`] = `
<DocumentFragment>
  <h1>
    Find your movie
  </h1>
  <div
    class="searchForm"
  >
    <input
      class="searchFormInput"
      placeholder="What do you want to watch?"
      type="text"
    />
    <button
      class="searchFormBtn"
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
